Assignment2-Python/sumLists.py

```diff
@@ -0,0 +1,92 @@
+#2.5 Sum Lists (p. 95)
+
+class Node:
+    def __init__(self, data=0, next=None):
+        self.data = data
+        self.next = next
+
+def print_list(node):
+    while node:
+        print(node.data, end=" -> ")
+        node = node.next
+    print("None")
+
+def sum_lists_reverse(l1, l2):
+    carry = 0
+    dummy_head = Node()
+    current = dummy_head
+
+    while l1 or l2 or carry:
+        total = carry
+        if l1:
+            total += l1.data
+            l1 = l1.next
+        if l2:
+            total += l2.data
+            l2 = l2.next
+        carry = total // 10
+        current.next = Node(total % 10)
+        current = current.next
+
+    return dummy_head.next
+
+def get_length(node):
+    length = 0
+    while node:
+        length += 1
+        node = node.next
+    return length
+
+def pad_list(node, padding):
+    head = node
+    for _ in range(padding):
+        new_node = Node(0)
+        new_node.next = head
+        head = new_node
+    return head
+
+def sum_helper(l1, l2):
+    if not l1 and not l2:
+        return 0, None
+
+    carry, next_node = sum_helper(l1.next, l2.next)
+    total = l1.data + l2.data + carry
+    current = Node(total % 10)
+    current.next = next_node
+    return total // 10, current
+
+def sum_lists_forward(l1, l2):
+    len1, len2 = get_length(l1), get_length(l2)
+    
+    if len1 < len2:
+        l1 = pad_list(l1, len2 - len1)
+    else:
+        l2 = pad_list(l2, len1 - len2)
+
+    carry, result = sum_helper(l1, l2)
+    
+    if carry:
+        return Node(carry, result)
+    return result
+
+
+l1_forward = Node(6, Node(1, Node(7)))  # Represents 617 in forward order
+l2_forward = Node(2, Node(9, Node(5)))  # Represents 295 in forward order
```

Comment:
`l1` and `l2` are still not good variable names, even if you append more to them.